deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvStages.java

```diff
@@ -103,6 +103,9 @@ public void update(@Context SecurityContext sc,
         if (origBean.getStage_type() != EnvType.DEFAULT && origBean.getStage_type() != environBean.getStage_type()) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Modification of stage type is not allowed!");
         }
+        if(environBean.getIs_sox() && origBean.getIs_sox() != environBean.getIs_sox()) {
+            authorizer.authorize(sc, new Resource(environBean.getEnv_name(), Resource.Type.ENV), Role.SYSTEMADMIN);
```

Comment:
I think I need to use the ENV/USER pair `SYSTEM/ADMIN` similar to this PR: https://github.com/pinterest/teletraan/blob/78cba3b8aa507408ab8812e81b4235656c20d807/deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/DeployCandidates.java#L52

In the database, we can then add corresponding users:
```
+-----------+-------------+---------------+-------+
| user_name | resource_id | resource_type | role  |
+-----------+-------------+---------------+-------+
| xxxxxxxx    | *           | SYSTEM        | ADMIN |
```